### DISCUSSION TITLE
SDK: How to get relational field

### DISCUSSION THREAD

**Discussion by emulienfou** [directus/directus#5499] [Archive]
Hello there,
I have an article who has a **Many to One** `category` field relation to a model `categories`.

Using the JS SDK, I retrieved an item using the next code:
```js
directus.items('articles').read({
      filter: {
        slug: { _eq: 'test' },
        status: { _eq: 'published' }
      }
    })
```
This will return some data like:
```js
{
   title: 'test',
   category: 3
}
```

However, how can I return a relational deep fields of my category, would like to have the field `title` instead of the category ID.
I saw there is a documentation for the REST API to query fields: https://docs.directus.io/reference/api/query/#fields

I'm searching to have something like:
```js
{
   title: 'test',
   category: {
      id: 3,
      title: 'Test'
   }
}
```
Tried the next code, but seems to not working at all:
```js
directus.items('articles').read({
      filter: {
        slug: { _eq: params.slug },
        status: { _eq: 'published' }
      },
      fields: ['*,category.*']
    })
```

**Comment by rijkvanzanten** [comment]
Try `fields: ['*', 'category.*']` 🙂

**Reply by emulienfou** [reply]
Just tried it, same issue: `category: 3`

**Reply by rijkvanzanten** [reply]
Does your user have read access to this category collection?

**Reply by rijkvanzanten** [reply]
Also make sure you're up to date (I believe `directus.items('articles').read({` was deprecated in the SDK a couple releases ago).

@WoLfulus any other insight here?

**Reply by emulienfou** [reply]
Yes, I have put both `articles` and `categories` in Read permissions for public

**Reply by emulienfou** [reply]
Using the package `@pnodev/nuxt-directus` for Nuxt.js

**Reply by WoLfulus** [reply]
@rijkvanzanten that package is using the previous SDK, still, what he mentioned should have worked. I haven't seen the implementation or if there's any middle steps done in that package though.

**Reply by emulienfou** [reply]
@WoLfulus @rijkvanzanten Now using the latest SDK version, without the Nuxt plugin.
I was able to make the `fields: ['*', 'category.*']` work perfectly.

However, I saw that the `fields` parameter is only available when using `readOne` but not seems to be available with `readMany`.   
So the **QueryOne** has a option `fields` but **QueryMany** did not!

In my case, you can I get a category relation without looping into all items?

**Reply by WoLfulus** [reply]
Are you sure? QueryMany is an intersection between QueryOne and many fields, so it should be there.

https://github.com/directus/directus/blob/main/packages/sdk/src/items.ts#L38

**Reply by emulienfou** [reply]
Never mind, I did not see the intersection, my bad.   
Working find actually, problem resolved
